### PR TITLE
Redux in React (PR)

### PR DESCRIPTION
**Pull request by Doheera-kosi**
# Milestone reach:

## General requirements
- No linter errors
- Used correct flow [Gitflow]
- Applied professional documentation skills

## HTML/CSS requirements
- Follow our list of the best practices for HTML & CSS
## JavaScript requirements
- Follow our list of [best practices for JavaScript]
## Project requirements
- Kept state immutable in reducers!
- Changed the default state in books reducer from an empty array to an array with a few books.
- Import Redux Provider and store in the main component of the React app.
- Connected app to the Redux store.
- Make sure that list of books displays books from Redux store.
- Use useSelector to consume the state.
- In your React component responsible for adding new books, add functionality for "Add new" button click:
- Import Redux useDispatch.
- Set the data inputs in the local React state set title and author
- Takes the inputs from the state and generate a unique id and add them to an object.

### CODE REVIEW

File: src/components/AddBook.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    dispatch(addBook(newBook))
+  }
+  return (
+    <div className='formContainer'>
+      <form className='form' onSubmit={handleSubmit}>
+        <label>Add a new book</label>
+        <input className='input' onChange={(e) => set_Title(e.target.value)} value={data_Title} type='text' placeholder='Title'/>
+        <input className='input' onChange={(e) => set_Author(e.target.value)} value={data_Author} type='text' placeholder='Author'/>
+        <button type='submit' className='submitBtn'>Submit</button>
+      </form>
```
Comment (Munsa1) [inline src/components/AddBook.js]: - [ ] Great work on imlementing the requirements for this project, but as at now the user is unable to add a new book.
image
- [ ] The submit button does not work, kindly fix it

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @Doheera-kosi, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow
## Required Changes ♻️
_Check the comments under the review._

- [ ] Kindly make your ReadMe file more descriptive by adding the relevant information
image
image


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/AddBook.js (hunk 1 above)
Comment (Doheera-kosi) [reply src/components/AddBook.js]: Thank you! Submit button now working properly

**Review by ichala — APPROVED** [review]
Hi @Doheera-kosi ,
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
 
68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966

# To Highlight 💪🏻

-  Changes Implemenetd✔️ 



 Good job implementing the requirements  :+1: Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Optional Suggestions  :

- N/A


**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag  @ichala in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**